src/backends/pyplot.jl

```diff
@@ -1317,13 +1317,13 @@ function py_add_legend(plt::Plot, sp::Subplot, ax)
                     elseif series[:seriestype] in (:path, :straightline, :scatter)
                         PyPlot.plt."Line2D"((0,1),(0,0),
                             color = py_color(single_color(get_linecolor(series, clims)), get_linealpha(series)),
-                            linewidth = py_thickness_scale(plt, clamp(get_linewidth(series), 0, 5)),
+                            linewidth = py_thickness_scale(plt, clamp(sp[:legendfontsize]/10, 0, 5)),
                             linestyle = py_linestyle(:path, get_linestyle(series)),
                             marker = py_marker(_cycle(series[:markershape], 1)),
-                            # markersize = py_thickness_scale(plt, series[:markersize]), # In case we decide that markersize needs to be scaled in the legend too
+                            markersize = py_thickness_scale(plt, sp[:legendfontsize]),
                             markeredgecolor = py_color(single_color(get_markerstrokecolor(series)), get_markerstrokealpha(series)),
                             markerfacecolor = py_color(single_color(get_markercolor(series, clims)), get_markeralpha(series)),
-                            markeredgewidth = py_thickness_scale(plt, series[:markerstrokewidth])
+                            markeredgewidth = py_thickness_scale(plt, series[:markerstrokewidth] * sp[:legendfontsize] / series[:markersize])   # retain the markersize/markerstroke ratio from the markers on the plot
```

Comment:
actually how about `0.8 * sp[:legendfontsize]`? I'm not sure how this looks with multiple series.

Reply:
I actually like the idea, but I also plot zero width markers from time to time and that would cause trouble here.

Reply:
Oh, sorry, my comment was meant for line 1323.

Reply:
```suggestion
                            markeredgewidth = py_thickness_scale(plt, 0.8 * series[:markerstrokewidth] * sp[:legendfontsize] / series[:markersize])   # retain the markersize/markerstroke ratio from the markers on the plot
```